MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,404 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Xna.Framework
+{
+    public static partial class MathHelper
+    {
+        /// <summary>
+        /// Class for generating random numbers consistantly when using Mono or Microsoft compilers
+        /// </summary>
+        public class Random
+        {
+            #region Static methods
+
+            [ClsCompliant(false)]
+            public static ulong GenerateSeed()
+            {
+                byte[] guid = Guid.NewGuid().ToByteArray();
+                return (BitConverter.ToUInt64(guid, 0) ^ BitConverter.ToUInt64(guid, 7));
+            }
+
+            #endregion
+
+            #region Private Fields
+
+            private ulong _seed;
+            private ulong _state;
+
+            #endregion
+
+            #region Properties
+
+            [ClsCompliant(false)]
+            public ulong Seed
+            {
+                get
+                {
+                    return this._seed;
+                }
+                set
+                {
+                    this._seed = value;
+                    this._state = value;
+                }
+            }
+
+            [ClsCompliant(false)]
+            public ulong State
+            {
+                get
+                {
+                    return this._state;
+                }
+            }
+
+            #endregion
+
+            #region Constructors
+
+            /// <summary>
+            /// Initializes a new instance of the random number generator using a new GUID as a default seed value
```

Comment:
We don't need to call out GUID in the docs.  Please remove mention of it.

This is an internal detail we could change based on platform or other needs that doesn't need to be public.